screens/Home/HomeScreen.tsx

```diff
@@ -11,153 +11,98 @@ import { useSafeAreaInsets } from "react-native-safe-area-context"
 import EventCard from "../../components/EventCard/EventCard"
 import { globalStyles } from "../../assets/global/globalStyles"
 import { useNavigation } from "@react-navigation/native"
-import React from "react"
+import React, { useEffect } from "react"
+import { Event } from "../../types/Event"
+import { getAllPastEvents, getAllFutureEvents } from "../../firebase/ManageEvents"
+import { showErrorToast } from "../../components/ToastMessage/toast"
 // will need to import it from the database later
 
-export interface event {
-  title: string
-  location: string
-  latitude: number
-  longitude: number
-  description: string
-  date: string
-  imageUrl: string
-}
-
-
-const events = [
-  {
-    title: "Balelek 2023",
-    location: "EPFL, Agora",
-    latitude: 46.51858962578904,
-    longitude: 6.566048509782951,
-    description: "Music festival 2023",
-    date: "2023-04-04", // changed to YYYY-MM-DD format
-    imageUrl: "https://balelec.ch/uploads/2023_f4caf642b6.jpg",
-  },
-  {
-    title: "Event 2",
-    location: "EPFL, CM",
-    latitude: 46.51859297638995, 
-    longitude: 6.563141941352757,
-    description: "Agepoly",
-    date: "2022-08-04", // changed to YYYY-MM-DD format
-    imageUrl: "https://example.com/image.png",
-  },
-  {
-    title: "Balelek 2024",
-    location: "EPFL, Agora",
-    latitude: 46.51992048733761,
-    longitude: 6.565926796074852,
-    description: "Music festival 2024",
-    date: "2024-05-20", // changed to YYYY-MM-DD format
-    imageUrl: "https://example.com/image.png",
-  },
-  {
-    title: "Graduation 2025",
-    location: "EPFL, Swisstech",
-    latitude: 46.52277976263842,
-    longitude: 6.565209257647138,
-    description: "Graduation",
-    date: "2025-06-04", // changed to YYYY-MM-DD format
-    imageUrl: "https://example.com/image.png",
-  },
-]
 
 const HomeScreen = () => {
   const insets = useSafeAreaInsets()
   const navigation = useNavigation()
-  //const [title, setTitle] = React.useState("Future")
-
-  const [searchQuery, setSearchQuery] = React.useState("")
-  // Filter events based on the current date and search query
-  const getFilteredEvents = (isFutureEvent: boolean) => {
-    const currentDate = new Date("2024-04-20")
+  const [futureEvents, setFutureEvents] = React.useState<Event[] | null>([])
+  const [pastEvents, setPastEvents] = React.useState<Event[] | null>([])
 
-    // Filter events based on date
-    const eventsFilteredByDate = events.filter((event) => {
-      const eventDate = new Date(event.date)
-      return isFutureEvent ? eventDate >= currentDate : eventDate < currentDate
-    })
+  const [ setSearchQuery] = React.useState("")
 
-    // Further filter by search query if there is one
-    return eventsFilteredByDate.filter((event) =>
-      event.title.toLowerCase().includes(searchQuery.toLowerCase())
-    )
+  const handleSearch = (search: string) => {
+    setSearchQuery(search)
   }
+  useEffect(() => {
```

Comment:
I believe you can use useFocusEffect() here https://reactnavigation.org/docs/use-focus-effect/ What it would do is reload the events every time this screen comes into focus. On the other hand the operation can be expensive so I'm just saying in case you didn't know. The decision to use it or not comes to you. You could also imagine to asynchronously check if there are new events on focus and update if necessary but not reload all events every time...

Reply:
I will definetly look into it